code/obj/artifacts/artifact_objects/heatwave.dm

```diff
@@ -0,0 +1,34 @@
+
+/obj/artifact/heatwave
+	name = "artifact heatwave"
+	associated_datum = /datum/artifact/heatwave
+
+/datum/artifact/heatwave
+	associated_object = /obj/artifact/heatwave
+	type_name = "Heat Surge"
+	rarity_weight = 365
+	validtypes = list("ancient","martian","eldritch","precursor")
```

Comment:
For reasons of theme, I would maybe consider replacing the martian origin here with a wizard origin. Making a big fireball seems pretty wizardy, and martian artifacts are damaged by heat (even if that probably won't work with a fireball due to jank code), so it seems they'd be unlikely to make something like this.